src/v/net/tls_certificate_probe.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/bytes.h"
+#include "config/tls_config.h"
+#include "seastarx.h"
+#include "ssx/metrics.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/net/tls.hh>
+
+#include <chrono>
+#include <exception>
+#include <iosfwd>
+
+namespace net {
+
+class tls_certificate_probe {
+    using tls_serial_number
+      = named_type<uint32_t, struct tls_serial_number_tag>;
+
+public:
+    using clock_type = ss::lowres_system_clock;
+    tls_certificate_probe() { reset(); }
+    tls_certificate_probe(const tls_certificate_probe&) = delete;
+    tls_certificate_probe& operator=(const tls_certificate_probe&) = delete;
+    tls_certificate_probe(tls_certificate_probe&&) = delete;
+    tls_certificate_probe& operator=(tls_certificate_probe&&) = delete;
+    ~tls_certificate_probe() = default;
+
+    void loaded(
+      const ss::tls::certificate_credentials& creds, std::exception_ptr ex);
+
+    void setup_metrics(std::string_view area, std::string_view detail);
+
+private:
+    ssx::metrics::all_metrics_groups _metrics;
+    std::chrono::seconds _load_time;
+    std::chrono::seconds _cert_expiry_time;
+    std::chrono::seconds _ca_expiry_time;
+    tls_serial_number _cert_serial;
+    tls_serial_number _ca_serial;
+    bool _cert_valid{false};
```

Comment:
nit: shoudl this be `_cert_loaded`?  I thought this was the flag to indicate if the certificate was valid (e.g. within start/end time frames, etc).